lib/wibox/drawable.lua

```diff
@@ -408,6 +410,11 @@ function drawable.new(d, widget_context_skeleton, drawable_name)
         ret:draw()
     end
 
+    d:connect_signal("collect", function()
+        print("\n\nCOLLECT!")
+        ret._redraw_pending = false
```

Comment:
I'd suggest `ret.was_GCd = true` and in the `:draw()` and `:_do_redraw()`-functions then `if ret.was_GCd then print("redraw after GC caused by", debug.traceback()) end`.